kemdk/src/main/java/dk/gls/kemdk/DeviceSerialUtil.kt

```diff
@@ -0,0 +1,161 @@
+                                }
+                            }
+                        }
+
+                        override fun onClosed() {
+                            Log.d("kemdk", "onClosed")
+                        }
+
+                        override fun onData(resultData: ProfileManager.ResultData?) {
+                            val result = resultData?.result
+                            if (result?.statusCode == EMDKResults.STATUS_CODE.CHECK_XML) {
+                                Log.d(
+                                    "kemdk",
+                                    "statusCode ${EMDKResults.STATUS_CODE.CHECK_XML}, responseXml ${result.statusString}"
+                                )
+                                //TODO The documentation says 'Parse the XML in result.statusString to know the status of profiles applied'. Do we want to parse the xml?
+                                //I think we have configured EMDK correctly by now, let's try and retrieve the serial
+                                val deviceSerial = retrieveDeviceSerial()
+                                Log.d("kemdk", "onData: retrieveDeviceSerial: $deviceSerial")
+                                if (continuation.isActive) {
+                                    if (deviceSerial != null && !deviceSerial.contains("Error")) {
+                                        continuation.resume(deviceSerial.toGLSSuccess())
+                                    } else {
+                                        continuation.resume(
+                                            EMDKThrowable.UnableToRetrieveSerial(
+                                                deviceSerial
+                                            ).toGLSFailure()
+                                        )
+                                    }
+                                }
+                            } else if (result?.statusCode != EMDKResults.STATUS_CODE.SUCCESS) {
+                                //Error occurred in applying the profile.
+                                Log.d(
+                                    "kemdk",
+                                    "statusCode error: ${result?.statusCode} ${result?.extendedStatusMessage}"
+                                )
+                                if (continuation.isActive) {
+                                    continuation.resume(
+                                        EMDKThrowable.ProfileXMLThrowable(result).toGLSFailure()
+                                    )
+                                }
+                            }
+                        }
+                    })
+
+                Log.d("kemdk", "getEMDKManager: ${emdkManagerResults.statusCode}")
+            }
+        }
+
+    /** Retrieve device serial. Be aware that the device should either be configured using Stage Now or above [retrieveAndConfigureAccessToDeviceSerial] function at least once before it is possible to retrieve the serial number **/
```

Comment:
I am not sure this creates java doc?

Reply:
It does. Would you prefer a multiline javadoc instead?

Reply:
No its fine then. 🙂